engine/access/rpc/backend/backend_stream_transactions_test.go

```diff
@@ -0,0 +1,365 @@
+	s.addNewFinalizedBlock(s.finalizedBlock.Header)
+	checkNewSubscriptionMessage(sub, flow.TransactionStatusExecuted)
+
+	// 4. Make the transaction block sealed, and add a new finalized block
+	s.sealedBlock = s.finalizedBlock
+	s.addNewFinalizedBlock(s.sealedBlock.Header)
+	checkNewSubscriptionMessage(sub, flow.TransactionStatusSealed)
+
+	// 5. Stop subscription
+	sub.Close()
+
+	// Ensure subscription shuts down gracefully
+	unittest.RequireReturnsBefore(s.T(), func() {
+		v, ok := <-sub.Channel()
+		assert.Nil(s.T(), v)
+		assert.False(s.T(), ok)
+		assert.NoError(s.T(), sub.Err())
+	}, 100*time.Millisecond, "timed out waiting for subscription to shutdown")
+}
+
+// TestSubscribeTransactionStatusExpired tests the functionality of the SubscribeTransactionStatuses method in the Backend
+// when transaction become expired
+func (s *TransactionStatusSuite) TestSubscribeTransactionStatusExpired() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	s.blocks.On("GetLastFullBlockHeight").Return(func() (uint64, error) {
+		return s.sealedBlock.Header.Height, nil
+	}, nil)
+
+	// Generate sent transaction with ref block of the current finalized block
+	transaction := unittest.TransactionFixture()
+	transaction.SetReferenceBlockID(s.finalizedBlock.ID())
+	txId := transaction.ID()
+
+	expectedMsgIndexCounter := counters.NewMonotonousCounter(0)
+
+	// Create a special common function to read subscription messages from the channel and check converting it to transaction info
+	// and check results for correctness
+	checkNewSubscriptionMessage := func(sub subscription.Subscription, expectedTxStatus flow.TransactionStatus) {
+		unittest.RequireReturnsBefore(s.T(), func() {
+			v, ok := <-sub.Channel()
+			require.True(s.T(), ok,
+				"channel closed while waiting for transaction info:\n\t- txID %x\n\t- blockID: %x \n\t- err: %v",
+				txId, s.finalizedBlock.ID(), sub.Err())
+
+			txInfo, ok := v.(*convert.TransactionSubscribeInfo)
+			require.True(s.T(), ok, "unexpected response type: %T", v)
+
+			assert.Equal(s.T(), txId, txInfo.ID)
+			assert.Equal(s.T(), expectedTxStatus, txInfo.Status)
+
+			expectedMsgIndex := expectedMsgIndexCounter.Value()
+			assert.Equal(s.T(), expectedMsgIndex, txInfo.MessageIndex)
+			wasSet := expectedMsgIndexCounter.Set(expectedMsgIndex + 1)
+			require.True(s.T(), wasSet)
+		}, time.Second, fmt.Sprintf("timed out waiting for transaction info:\n\t- txID: %x\n\t- blockID: %x", txId, s.finalizedBlock.ID()))
+	}
+
+	// Subscribe to transaction status and receive the first message with pending status
+	sub := s.backend.SubscribeTransactionStatuses(ctx, &transaction.TransactionBody)
+	checkNewSubscriptionMessage(sub, flow.TransactionStatusPending)
+
+	// Generate 600 blocks without transaction included and check, that transaction still pending
+	startHeight := s.finalizedBlock.Header.Height + 1
+	lastHeight := startHeight + flow.DefaultTransactionExpiry + 1
+
+	for i := startHeight; i < lastHeight; i++ {
```

Comment:
nit: `lastHeight` implies it is the last height visited
```suggestion
	lastHeight := startHeight + flow.DefaultTransactionExpiry

	for i := startHeight; i <= lastHeight; i++ {
```